Add method to 'trigger' QgsTasks

nyalldawson opened the pull request:
Triggering occurs when a task is clicked in the task manager widget, and this can be used to e.g. open a dialog showing detailed task progress (or reopen a closed dialog which started the task)

nyalldawson committed:
Add method to 'trigger' QgsTasks

Triggering occurs when a task is clicked in the task manager
widget, and this can be used to e.g. open a dialog showing
detailed task progress (or reopen a closed dialog which started
the task)

nyalldawson committed:
When processing tasks are clicked in task manager, reopen the
algorithm dialog

m-kuhn commented:
I'm 50/50 on how to do this. Trigger is a quite generic name, unspecific to what is expected if issued.
Still it's sent through the manager and not specific tasks, hence generic handling is expected.
I'd say at least one of the two should be adjusted.

nyalldawson committed:
Rename QgsTaskManager::triggered to taskTriggered

nyalldawson committed:
Show display name, not id, for background algorithm tasks

nyalldawson committed:
Improved memory managment of algorithm dialogs

Ensure that dialogs are always correctly deleted when appropriate.

Also, if an algorithm is running in a background task and hits
an error, we automatically re-show the algorithm dialog and the
associated log for debugging.

Fixes #16858

nyalldawson committed:
Fix build

nyalldawson committed:
Sipify
